### PR TITLE
feat(secrets): Call secrets verify API

### PR DESCRIPTION
**Pull request by Eliran-Turgeman**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

* calling secrets verifier api
* adding status to the record and printing it

*Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.*

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by arielkru — APPROVED** [review]
Nice :)

**Review by gruebel — COMMENTED** [review]
noice 🍻

File: checkov/common/bridgecrew/platform_integration.py [hunk 1]
```diff
@@ -411,6 +412,16 @@ def persist_image_scan_results(self, report: dict[str, Any] | None, file_path: s
         to_upload = {"report": report, "file_path": file_path, "image_name": image_name, "branch": branch}
         _put_json_object(self.s3_client, to_upload, self.bucket, target_report_path)
 
+    def persist_enriched_secrets(self, enriched_secrets: list[EnrichedSecret]) -> str | None:
+        if not enriched_secrets or not self.repo_path or not self.bucket:
+            logging.debug('One of enriched secrets, repo path, or bucket are empty, aborting')
```
[the remark this answers is not in the document]
Comment (Eliran-Turgeman) [reply checkov/common/bridgecrew/platform_integration.py]: done

File: checkov/common/util/time_utils.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+def timeit(func: Any) -> Any:
+    @wraps(func)
+    def timeit_wrapper(*args: Any, **kwargs: Any) -> Any:
+        start_time = time.perf_counter()
+        result = func(*args, **kwargs)
+        end_time = time.perf_counter()
+        total_time = end_time - start_time
+        logging.info(f'Function {func.__name__} Took {total_time:.4f} seconds')
+        return result
+    return timeit_wrapper
```
[the remark this answers is not in the document]
Comment (Eliran-Turgeman) [reply checkov/common/util/time_utils.py]: Looking better without all the `Any` typing 😄

File: checkov/common/util/time_utils.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import logging
```
[the remark this answers is not in the document]
Comment (Eliran-Turgeman) [reply checkov/common/util/time_utils.py]: decorators sound good

File: checkov/common/output/record.py [hunk 4]
```diff
@@ -194,6 +194,9 @@ def to_string(self, compact: bool = False, use_bc_ids: bool = False) -> str:
 
         status_message = colored("\t{} for resource: {}\n".format(status, self.resource), status_color)
         if self.check_result["result"] == CheckResult.FAILED and code_lines and not compact:
+            if hasattr(self, 'validation_status'):
+                validation_status_string = f'\tStatus: {self.validation_status}\n'
+                return f"{check_message}{status_message}{severity_message}{detail}{file_details}{validation_status_string}{caller_file_details}{guideline_message}{code_lines}{evaluation_message}"
```
[the remark this answers is not in the document]
Comment (Eliran-Turgeman) [reply checkov/common/output/record.py]: moved the validation_status_string above the if-if-else block

File: checkov/common/output/secrets_record.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+from checkov.common.bridgecrew.severities import Severity
+
+from checkov.common.output.record import Record
+from checkov.common.typing import _CheckResult
+
+
+class SecretsRecord(Record):
+    def __init__(self,
+                 validation_status: Optional[str],
```
[the remark this answers is not in the document]
Comment (Eliran-Turgeman) [reply checkov/common/output/secrets_record.py]: done

File: checkov/secrets/consts.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
+from enum import Enum
+
+
+class ValidationStatus(Enum):
+    PRIVILEGED = 'Privileged'
+    VALID = 'Valid'
+    INVALID = 'Invalid'
+    UNKNOWN = 'Unknown'
```
[the remark this answers is not in the document]
Comment (Eliran-Turgeman) [reply checkov/secrets/consts.py:8]: done

File: checkov/secrets/runner.py [hunk 7]
```diff
@@ -191,7 +200,14 @@ def run(
...
                     evaluations=None,
                     file_abs_path=os.path.abspath(secret.filename),
+                    validation_status=ValidationStatus.UNKNOWN.value
                 ))
+
+            enriched_secrets_s3_path = bc_integration.persist_enriched_secrets(self.secrets_coordinator.get_secrets())
+            if not enriched_secrets_s3_path:
+                return report
+
+            self.verify_secrets(report, enriched_secrets_s3_path)
```
[the remark this answers is not in the document]
Comment (Eliran-Turgeman) [reply checkov/secrets/runner.py]: Done

File: checkov/secrets/runner.py [hunk 8]
```diff
@@ -260,8 +277,61 @@ def search_for_suppression(
...
-                                   result: _CheckResult)\
+                                   result: _CheckResult) \
             -> None:
         if result.get('result') == CheckResult.FAILED and secret_value is not None:
             enriched_secret = EnrichedSecret(original_secret=secret_value, bc_check_id=bc_check_id, resource=resource)
             self.secrets_coordinator.add_secret(enriched_secret=enriched_secret)
+
+    @timeit
+    def verify_secrets(self, report: Report, enriched_secrets_s3_path: str) -> VerifySecretsResult:
+        if not convert_str_to_bool(os.getenv("CKV_VALIDATE_SECRETS", False)) or not os.getenv("BC_API_KEY"):
```
[the remark this answers is not in the document]
Comment (Eliran-Turgeman) [reply checkov/secrets/runner.py]: Done

File: checkov/secrets/runner.py [hunk 9]
```diff
@@ -260,8 +277,61 @@ def search_for_suppression(
...
+        verification_report_presigned_url = response.json()['verificationReportSignedUrl']
+        verification_report = self.get_json_verification_report(verification_report_presigned_url)
+
+        if not verification_report:
+            return VerifySecretsResult.FAILURE
+
+        validation_status_by_check_id_and_resource = {}
+        for validation_status_entity in verification_report:
+            key = f'{validation_status_entity.get("violationId")}_{validation_status_entity.get("resourceId")}'
+            validation_status_by_check_id_and_resource[key] = validation_status_entity.get('status')
```
[the remark this answers is not in the document]
Comment (Eliran-Turgeman) [reply checkov/secrets/runner.py]: add a check to all keys

**Review by gruebel — APPROVED** [review]
🍻

**Pull request merged**